### PR TITLE
mixed precision exchange

### PR DESCRIPTION
**Pull request by mcallic2**
**Description**
All real values will be calculated as r8_kind

Fixes # (issue)

**How Has This Been Tested?**
Autotools with intel and gcc

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

File: .github/workflows/Dockerfile.gnu [hunk 1]
```diff
@@ -0,0 +1,68 @@
+#***********************************************************************
```
Comment (mlee03) [inline .github/workflows/Dockerfile.gnu:1]: should .github be updated to that in mixedmode?

**Review by mlee03 — COMMENTED** [review]
@mcallic2 can you merge in mixedmode?

File: exchange/xgrid.F90 [hunk 2]
```diff
@@ -774,7 +773,7 @@ subroutine load_xgrid (xmap, grid, grid_file, grid1_id, grid_id, tile1, tile2, u
         call get_variable_units(fileobj, "xgrid_area", attvalue)
 
         if( trim(attvalue) == 'm2' ) then
-           garea = 4.0*PI*RADIUS*RADIUS;
+           garea = 4.0_r8_kind * real(PI,r8_kind) * real(RADIUS,r8_kind) * real(RADIUS, r8_kind);
```
Comment (mlee03) [inline exchange/xgrid.F90:776]: if garea is already 48, then PI and RADIUS do not need real()

File: exchange/xgrid.F90 [hunk 3]
```diff
@@ -1310,10 +1309,10 @@ subroutine get_grid_version1(grid, grid_id, grid_file)
   real(r8_kind), dimension(grid%im) :: lonb
   real(r8_kind), dimension(grid%jm) :: latb
   real(r8_kind)                     :: d2r
-  integer                  :: is, ie, js, je
-  type(FmsNetcdfDomainFile_t) :: fileobj
+  integer                           :: is, ie, js, je
+  type(FmsNetcdfDomainFile_t)       :: fileobj
 
-  d2r = PI/180.0
+  d2r = real(PI, r8_kind) / 180.0_r8_kind
```
Comment (mlee03) [inline exchange/xgrid.F90:1315]: likewise, do not need real() for PI

File: exchange/xgrid.F90 [hunk 4]
```diff
@@ -1375,17 +1374,17 @@ subroutine get_grid_version2(grid, grid_id, grid_file)
...
+  integer                    :: is, ie, js, je, nlon, nlat, i, j
+  integer                    :: start(4), nread(4), isc2, iec2, jsc2, jec2
+  type(FmsNetcdfFile_t)      :: fileobj
 
   if(.not. open_file(fileobj, grid_file, 'read') ) then
      call error_mesg('xgrid_mod(get_grid_version2)', 'Error in opening file '//trim(grid_file), FATAL)
   endif
 
-  d2r = PI/180.0
+  d2r = real(PI, r8_kind) / 180.0_r8_kind
```
Comment (mlee03) [inline exchange/xgrid.F90:1387]: and here

File: exchange/xgrid.F90 [hunk 5]
```diff
@@ -4352,8 +4351,8 @@ function grad_zonal_latlon(d, lon, lat, is, ie, js, je, isd, jsd)
      endif
      dx = lon(ip1) - lon(im1)
      if(abs(dx).lt.EPS )  call error_mesg('xgrids_mod(grad_zonal_latlon)', 'Improper grid size in lontitude', FATAL)
-     if(dx .gt. PI)  dx = dx - 2.0* PI
-     if(dx .lt. -PI) dx = dx + 2.0* PI
+     if(dx .gt. real(PI,r8_kind))  dx = dx - 2.0_r8_kind* real(PI, r8_kind)
```
Comment (mlee03) [inline exchange/xgrid.F90:4354]: here too

File: test_fms/exchange/test_xgrid.F90 [hunk 6]
```diff
@@ -459,14 +459,14 @@ program xgrid_test
         xt = 0; yt = 0;
         do j = jsc_atm, jec_atm
            do i = isc_atm, iec_atm
-              xt(i,j) = tmpx(2*i, 2*j)*DEG_TO_RAD
-              yt(i,j) = tmpy(2*i, 2*j)*DEG_TO_RAD
+              xt(i,j) = tmpx(2*i, 2*j)*real(DEG_TO_RAD, r8_kind)
```
Comment (mlee03) [inline test_fms/exchange/test_xgrid.F90:462]: here too

**Comment by mcallic2** [comment]
> @mcallic2 can you merge in mixedmode?

@mlee03  this is up to date with mixedmode

File: .github/workflows/Dockerfile.gnu (hunk 1 above)
Comment (mcallic2) [reply .github/workflows/Dockerfile.gnu:1]: this may have been me messing up the mixedmode merge

File: exchange/xgrid.F90 (hunk 2 above)
Comment (mcallic2) [reply exchange/xgrid.F90:776]: hmm, i wonder if I should go back to astronomy_mod and make some more changes if thats the case then

File: .github/workflows/Dockerfile.gnu (hunk 1 above)
Comment (mlee03) [reply .github/workflows/Dockerfile.gnu:1]: @mcallic2 this

**Comment by mcallic2** [comment]
./github directory is bugging me

**Pull request closed without merging**